source/Camera/Basic/C++Halcon/CaptureViaGenICam/CaptureViaGenICam.cpp

```diff
@@ -6,92 +6,116 @@
-	objectModel3D.SetObjectModel3dAttribMod(HalconCpp::HTuple("blue"), "points", RGB.AccessChannel(3).GetGrayval(rows, cols));
+    objectModel3D.SetObjectModel3dAttribMod(HalconCpp::HTuple("red"),
+                                            "points",
+                                            RGB.AccessChannel(1).GetGrayval(rows, cols));
+    objectModel3D.SetObjectModel3dAttribMod(HalconCpp::HTuple("green"),
+                                            "points",
+                                            RGB.AccessChannel(2).GetGrayval(rows, cols));
+    objectModel3D.SetObjectModel3dAttribMod(HalconCpp::HTuple("blue"),
+                                            "points",
+                                            RGB.AccessChannel(3).GetGrayval(rows, cols));
 }
 
-HalconCpp::HString getFirstAvailableZividDevice(const HalconCpp::HTuple &devices)
+HalconCpp::HString getFirstAvailableZividDevice()
 {
-	auto zividDevices = devices.TupleRegexpSelect("Zivid");
-	if (zividDevices.Length() == 0)
-	{
-		throw std::runtime_error("No Zivid devices found. Please check your setup.");
-	}
-	return zividDevices[0];
+    auto devices = HalconCpp::HTuple();
+    auto information = HalconCpp::HTuple();
+    HalconCpp::InfoFramegrabber("GenICamTL", "device", &information, &devices);
+
+    auto zividDevices = devices.TupleRegexpSelect("Zivid");
+    if(zividDevices.Length() == 0)
+    {
+        throw std::runtime_error("No Zivid devices found. Please check your setup.");
+    }
+    return zividDevices[0];
 }
 
 int main()
 {
-	try
-	{
-		auto availableDevices = HalconCpp::HTuple();
-		auto information = HalconCpp::HTuple();
-		HalconCpp::InfoFramegrabber("GenICamTL", "device", &information, &availableDevices);
-		const auto zividDevice = getFirstAvailableZividDevice(availableDevices);
-
-		std::cout << "Connecting to camera" << std::endl;
-		auto Framegrabber = HalconCpp::HTuple();
-		HalconCpp::OpenFramegrabber("GenICamTL", 1, 1, 0, 0, 0, 0, "progressive", -1, "default", -1, "false", "default", zividDevice, 0, 0, &Framegrabber);
+    try
+    {
+        std::cout << "Connecting to camera" << std::endl;
+        const auto zividDevice = getFirstAvailableZividDevice();
+        auto Framegrabber = HalconCpp::HTuple();
+        HalconCpp::OpenFramegrabber("GenICamTL",
+                                    1,
+                                    1,
+                                    0,
+                                    0,
+                                    0,
+                                    0,
+                                    "progressive",
+                                    -1,
+                                    "default",
+                                    -1,
+                                    "false",
+                                    "default",
+                                    zividDevice,
+                                    0,
+                                    0,
+                                    &Framegrabber);
```

Comment:
Can you ask someone from SE if this is expected from clang-tidy? It doesn't look good to me.

Reply:
I believe it's `clang-format` that does modifications to your formatting. `clang-tidy` only checks and gives you errors and warnings (can also be formatting)